contrib/node/src/python/pants/contrib/node/targets/node_module.py

```diff
@@ -24,10 +30,15 @@ def __init__(self, sources=None, address=None, payload=None, **kwargs):
     # tasks.  The reality is likely to be though that both pants will never cover all cases, and a
     # back door to execute new tools during development will be desirable and supporting conversion
     # of pre-existing package.json files as node_module targets will require this.
+    package_manager = 'yarnpkg' if package_manager == 'yarn' else package_manager
```

Comment:
I believe providing a good experience for users is very important and it's better than just failing developers' intuition. I personally forget to write yarnpkg frequently in the environment with hadoop yarn installed. If there's no restriction, I don't think we should restrict developers' mind.